provider/datagen/src/transform/segmenter/lstm.rs

```diff
@@ -84,3 +170,89 @@ impl IterableDataProvider<LstmDataV1Marker> for crate::DatagenProvider {
         ])
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use icu_provider_adapters::any_payload::AnyPayloadProvider;
+    use icu_provider_adapters::fork::ForkByKeyProvider;
+    use icu_segmenter::LineSegmenter;
+
+    fn load_lstm_data_directly(filename: &str) -> DataPayload<LstmDataV1Marker> {
+        let json_str = std::fs::read(filename).unwrap();
+        let raw_data: RawLstmData = serde_json::from_slice(&json_str).unwrap();
+        let data = raw_data.try_convert().unwrap();
+        DataPayload::from_owned(data)
+    }
+
+    #[test]
+    fn burmese_word_break() {
+        // "Burmese Language" in Burmese
+        const TEST_STR: &str = "မြန်မာဘာသာစကား";
+
+        let provider = crate::DatagenProvider::for_test();
+        let segmenter = LineSegmenter::try_new_lstm_unstable(&provider).unwrap();
+        let breaks: Vec<usize> = segmenter.segment_str(TEST_STR).collect();
+        // LSTM model breaks more characters, but it is better to return [30].
+        assert_eq!(breaks, [12, 18, 30, TEST_STR.len()], "Burmese test");
```

Comment:
Question: the new `assert_eq!` in these tests all have a break point at the end of the string, but the old ones didn't. Is this a expect behavior change, and why?